x-pack/filebeat/input/cel/input_test.go

```diff
@@ -1710,6 +1710,96 @@ var inputTests = []struct {
 			},
 		},
 	},
+	{
+		name: "file_auth_default_header",
+		server: func(t *testing.T, h http.HandlerFunc, config map[string]interface{}) {
+			dir := t.TempDir()
+			secret := "file-secret"
+			path := filepath.Join(dir, "auth_token")
+			if err := os.WriteFile(path, []byte(secret+"\n"), 0o600); err != nil {
+				t.Fatalf("failed to write auth token: %v", err)
+			}
```

Comment:
Do we gain anything here by using io/fs and testing/fstest for the underlying fs?